homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -194,22 +232,25 @@ async def _execute(
 class AudioPipelineRequest(PipelineRequest):
     """Request to full pipeline from audio input (stt) to audio output (tts)."""
 
-    intent_input: str  # this will be changed to stt audio
+    stt_metadata: stt.SpeechMetadata
+    stt_stream: AsyncIterable[bytes]
     conversation_id: str | None = None
 
     async def _execute(self, run: PipelineRun):
         run.start()
 
-        # stt will go here
+        stt_result = await run.speech_to_text(self.stt_metadata, self.stt_stream)
+        if stt_result.result == stt.SpeechResultState.SUCCESS:
```

Comment:
If not a success, `run.speech_to_text` should already have sent an error event and we should not finish the pipeline.
```python
        if stt_result.result != stt.SpeechResultState.SUCCESS:
            return
```